src/test/scala/it/pagopa/interop/partyprocess/ExternalApiSpec.scala

```diff
@@ -254,4 +256,131 @@ trait ExternalApiSpec
     }
 
   }
+
+  "Get institution/product billing data" must {
+    val productId = "prod-io"
+
+    val orgPartyId = UUID.randomUUID()
+    val originId   = UUID.randomUUID().toString
+    val externalId = UUID.randomUUID().toString
+    val origin     = "ORIGIN"
+
+    val institution = PartyManagementDependency.Institution(
+      id = orgPartyId,
+      externalId = externalId,
+      originId = originId,
+      description = "description",
+      digitalAddress = "digitalAddress",
+      attributes = Seq(PartyManagementDependency.Attribute(origin, "C17", "attrs")),
+      taxCode = "taxCode",
+      origin = origin,
+      address = "address",
+      zipCode = "zipCode",
+      products = Map(
+        "prod-io" -> PartyManagementDependency.InstitutionProduct(
+          product = "prod-io",
+          pricingPlan = Option("pricingPlan"),
+          billing = PartyManagementDependency
+            .Billing(vatNumber = "VATNUMBER", recipientCode = "RECIPIENTCODE", publicServices = Option(true))
+        )
+      )
+    )
+
+    val expected = BillingData(
+      institutionId = institution.id,
+      externalId = institution.externalId,
+      origin = institution.origin,
+      originId = institution.originId,
+      description = institution.description,
+      taxCode = institution.taxCode,
+      digitalAddress = institution.digitalAddress,
+      address = institution.address,
+      zipCode = institution.zipCode,
+      institutionType = institution.institutionType,
+      pricingPlan = Option("pricingPlan"),
+      billing = Billing(vatNumber = "VATNUMBER", recipientCode = "RECIPIENTCODE", publicServices = Option(true))
+    )
+
+    def mockPartyManagement(retrieveInstitution: Boolean, existsBilling: Boolean) = {
+      (mockPartyManagementService
+        .retrieveInstitutionByExternalId(_: String)(_: String)(_: Seq[(String, String)]))
+        .expects(externalId, *, *)
+        .returning(
+          if (retrieveInstitution)
+            Future.successful(
+              if (existsBilling) institution
+              else
+                institution.copy(products =
+                  Map(
+                    "prodX" -> InstitutionProduct(
+                      product = "prodX",
+                      pricingPlan = None,
+                      billing = PartyManagementDependency.Billing(
+                        vatNumber = "VATNUMBER",
+                        recipientCode = "RECIPIENTCODE",
+                        publicServices = Option(true)
+                      )
+                    )
+                  )
+                )
+            )
+          else Future.failed(ResourceNotFoundError(externalId))
```

Comment:
personal taste, but I'd suggest to keep the mocks simple to improve readability.
I would favour having multiple mock functions without parameters, suck as `mockRetrieveInstitutionNotFound`, `mockRetrieveInstitutionWithoutBilling` and `mockRetrieveInstitutionWithBilling`.

But it's really just a mild suggestion 😄

Reply:
this is a java vs scala developer question I think 🤣
I'l change it 😉